openedx/openedx-platform
# [ENT-365] Move from python-social-auth to split social

This PR aims to bump `edx-platform`'s use of the old, monolithic version of [`python-social-auth`](https://github.com/omab/python-social-auth) (PSA) to the latest, split version: [`social-app-django`](https://github.com/python-social-auth/social-auth-app-django) (v1.2.0) and [`social-auth-core`](https://github.com/python-social-auth/social-core) (v1.3.0).

**JIRA tickets**: Implements ENT-365.

**Discussions**: Upstream [PSA](https://github.com/python-social-auth) decided to break away from the [monolithic repo](https://github.com/omab/python-social-auth/) December '16. They're now under an organization and have also made some changes to functionality we'd like to have the `edx-platform` make use of, such as a [configurable maximum session length](https://github.com/python-social-auth/social-app-django/pull/54), and more.

**Dependencies**: None

**Sandbox URL**: [https://pr15135.sandbox.opencraft.hosting/](https://pr15135.sandbox.opencraft.hosting/)

**Deployment targets**: edx.org and edge.edx.org

**Merge deadline**: No tight deadline.

**Testing instructions**:

- OAuth2
	- Go to [https://pr15135.sandbox.opencraft.hosting/](https://pr15135.sandbox.opencraft.hosting/).
	- Click `REGISTER` on the top left.
	- Facebook
		- Click 'Facebook' and register by filling in any remaining necessary fields.
		- (optional) Verify the email.
		- Logout.
		- Press `Sign in` on the top right.
		- Sign in with Facebook.
	- Google+
		- Click 'Google+' and choose an account.
		- Register by filling in any remaining necessary fields.
		- (optional) Verify the email.
		- Logout.
		- Press `Sign in` on the top right.
		- Sign in with Google+.
		- Choose the account you registered with.
	- Linkedin
		- Click 'Linkedin' and register by filling in any remaining necessary fields.
		- (optional) Verify the email.
		- Logout.
		- Press `Sign in` on the top right.
		- Sign in with Linkedin.
- SAML IdP
	- TestShib
		- Click 'TestShib' and enter any of the username/password combinations available.
		- Fill in an email (whether you can verify or not) and complete registration.
		- Logout.
		- Press 'Sign in' on the top right.
		- Sign in with TestShib.
- LTI
	- [LTI Tools](http://lti.tools/test/tc.php)
		- Please ask for the consumer key/shared secret if you'd like to try this but don't have admin access.

**Author notes and concerns**:

1. Currently based off of @haikuginger's initial work.
2. This PR is owned by @UmanShahzad; @haikuginger opened it on his behalf to avoid the automatic OSPR process.
3. **This will have to get merged along with** https://github.com/edx/edx-enterprise/pull/120

**Settings**
```yaml
EDXAPP_FEATURES:
  ENABLE_COMBINED_LOGIN_REGISTRATION: true
  ENABLE_THIRD_PARTY_AUTH: true
SWAPFILE_SIZE: 4GB
```
